auth_helper.py

```diff
@@ -0,0 +1,68 @@
+import os
+import pathlib
+
+import json5
+import sys
+import requests
+import msal
+
+class AuthHelper:
+    tenantId = os.environ["tenant_id"]
+    scope = os.environ["scope"]
+    authority = os.environ["authority"]
+    appId = os.environ["appId"]
+    accessToken = ""
+    def __init__(self,**kwargs):
+        for key,value in kwargs.items():
+            setattr(self,key,value)
+
+    def adoAuthenticate(self):
+        mscache = msal.SerializableTokenCache()
+        output = pathlib.Path(__file__).parent / pathlib.Path("token.bin")
+        if os.path.exists(output):
+            print("Deserializing cached credentials.")
+            mscache.deserialize(open(output, "r").read())
```

Comment:
Encoding should be set for all `open` calls. Also, this file isn't closed. Using a context manager is the best option.

Reply:
Resolved, used 'utf-8', not sure if you care what is used.